src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -105,4 +133,192 @@ private static List<File> getTargetFiles(String path, String... moduleFolders)
+        final String description = SiteUtil.getDescriptionFromJavadoc(moduleJavadoc);
+
+        final List<ModulePropertyDetails> propertiesDetails = getPropertiesDetails(properties,
+                javadocs, className, instance);
+
+        return new ModuleDetails(moduleName, fullyQualifiedName, parentModuleString, description,
+                moduleType, propertiesDetails, new ArrayList<>(messageKeys));
+    }
+
+    /**
+     * Get module type(check/filter/filefilter) based on module name.
+     *
+     * @param moduleName module name.
+     * @return module type.
+     */
+    private static ModuleType getModuleType(String moduleName) {
+        final ModuleType result;
+        if (moduleName.endsWith("FileFilter")) {
+            result = ModuleType.FILEFILTER;
+        }
+        else if (moduleName.endsWith("Filter")) {
+            result = ModuleType.FILTER;
+        }
+        else {
+            result = ModuleType.CHECK;
+        }
+        return result;
+    }
+
+    /**
+     * Get property details for the given property - name, description, type, default value.
+     *
+     * @param properties properties of the module.
+     * @param javadocs javadocs of the module.
+     * @param className the class name of the module.
+     * @param instance the instance of the module.
+     * @return property details.
+     * @throws MacroExecutionException if an error occurs.
+     */
+    private static List<ModulePropertyDetails> getPropertiesDetails(
+            Set<String> properties, Map<String, DetailNode> javadocs,
+            String className, Object instance)
+            throws MacroExecutionException {
+        final List<ModulePropertyDetails> result = new ArrayList<>();
+        for (String property : properties) {
+            final String description = getPropertyDescription(property,
+                    javadocs.get(property));
+            final Field field = SiteUtil.getField(instance.getClass(), property);
+            final String type =
+                    SiteUtil.getFieldClass(field, property, className, instance).getCanonicalName();
+            final String defaultValue = getPropertyDefaultValue(property, field, instance,
+                    className);
+            final String validationType = getValidationType(property);
+
+            result.add(new ModulePropertyDetails(property, type, defaultValue,
+                    validationType, description));
+        }
+        return result;
+    }
+
+    /**
+     * Get validation type for the given property.
+     *
+     * @param property property name.
+     * @return validation type.
+     */
+    private static String getValidationType(String property) {
+        final String validationType;
+        if (SiteUtil.TOKENS.equals(property) || SiteUtil.JAVADOC_TOKENS.equals(property)) {
+            validationType = "tokenSet";
+        }
+        else {
+            validationType = null;
+        }
+        return validationType;
```

Comment:
We will have to transfer the `validationType` to the Javadoc. What about a Javadoc param?
>  * Validation type is {code tokenTypesSet}.
```
    /**
     * description...
     *
     * @param allow User's value of allowSamelineSingleParameterlessAnnotation.
     * @validationType tokenTypesSet
     * @since 6.1
     */
    public final void setAllowSamelineSingleParameterlessAnnotation(boolean allow) {
        allowSamelineSingleParameterlessAnnotation = allow;
    }
```

Reply:
Doesn't have to be in the scope of this PR.

Reply:
Please create separate issue on this and make example in read properly and validation type to let me understand

Reply:
For now I think this hardcod is enough.

Reply:
@stoyanK7 , your example is very confusing.
We use validation type for more complicated types, not a booleans 
https://github.com/checkstyle/checkstyle/blob/f58e45eb77487425249b7b0cc2b8bebe54662b51/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/annotation/AnnotationLocationCheck.xml#L59

Reply:
Only tokens https://github.com/search?q=repo%3Acheckstyle%2Fcheckstyle+validation-type&type=code&p=1

Reply:
Moving to https://github.com/checkstyle/checkstyle/issues/13722